### PR TITLE
Supervisor spring cleaning public fields

### PR DESCRIPTION
**Pull request by bodymindarts**
This PR makes all public fields and methods not used by clients private.
This in itself does not really improve the design of the code base but it makes it significantly easier to perform additional refactoring because it is easier to figure out what the effective public interface of the objects are.

See https://github.com/habitat-sh/habitat/issues/2094 for more context.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/butterfly/src/server/mod.rs [hunk 1]
```diff
@@ -393,27 +387,28 @@ impl Server {
...
+    //     let trace_health = health.clone();
+    //     if self.member_list.insert_health(member, health) {
+    //         trace_it!(MEMBERSHIP: self,
+    //                   TraceKind::MemberUpdate,
+    //                   trace_member_id,
+    //                   trace_incarnation,
+    //                   trace_health);
+    //         self.rumor_list.insert(rk);
+    //     }
+    // }
```
Comment (mwrock) [inline components/butterfly/src/server/mod.rs]: I'd be in favor of just deleting this dead block.

**Comment by mwrock** [comment]
Definitely think this is a step in the right direction.

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/221759889).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/221759889).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**